jaxrs/src/test/java/org/killbill/billing/jaxrs/resources/TestOverdueResource.java

```diff
@@ -0,0 +1,146 @@
+    private final PaymentApi paymentApi = mock(PaymentApi.class);
+    private final InvoicePaymentApi invoicePaymentApi = mock(InvoicePaymentApi.class);
+    private final OverdueApi overdueApi = mock(OverdueApi.class);
+    private final Clock clock = mock(Clock.class);
+    private final Context context = mock(Context.class);
+
+
+    private OverdueResource newOverdueResource() {
+        return new OverdueResource(
+                jaxrsUriBuilder,
+                tagUserApi, customFieldUserApi, auditUserApi, accountUserApi,
+                paymentApi, invoicePaymentApi, overdueApi,
+                clock, context);
+    }
+
+
+    protected DefaultOverdueConfig getOverdueConfig(final String name) throws Exception {
+        return XMLLoader.getObjectFromString(Resources.getResource("org/killbill/billing/jaxrs/resources/overdue/" + name).toExternalForm(), DefaultOverdueConfig.class);
+    }
+
+    /**
+     * Represent invalid overdue that cannot marshalled into JSON. This XML contains the same value as defined in:
+     * https://github.com/killbill/killbill/issues/1497
+     *
+     * This XML causing NullPointerException when JSON marshalling because it missing
+     * 'timeSinceEarliestUnpaidInvoiceEqualsOrExceeds' element.
+     */
+    @Test(groups = "fast", description = "https://github.com/killbill/killbill/issues/1497")
+    public void testGetOverdueConfigJson_noTimeSinceEarliest() throws Exception {
+        final OverdueConfig overdueConfig = getOverdueConfig("overdue_no-TimeSinceEarliest.xml");
+
+        when(context.createTenantContextNoAccountId(servletRequest)).thenReturn(tenantContext);
+        when(overdueApi.getOverdueConfig(any())).thenReturn(overdueConfig);
+
+        final OverdueResource resource = newOverdueResource();
+        final Response response = resource.getOverdueConfigJson(any());
+
+        Assert.assertEquals(response.getStatus(), 200);
+    }
+
+    /**
+     * Represent valid overdue configuration that can be marshalled into JSON.
+     */
+    @Test(groups = "fast")
+    public void testGetOverdueConfigJson() throws Exception {
+        final OverdueConfig overdueConfig = getOverdueConfig("overdue_valid.xml");
+
+        when(context.createTenantContextNoAccountId(servletRequest)).thenReturn(tenantContext);
+        when(overdueApi.getOverdueConfig(any())).thenReturn(overdueConfig);
+
+        final OverdueResource resource = newOverdueResource();
+        final Response response = resource.getOverdueConfigJson(any());
+
+        Assert.assertEquals(response.getStatus(), 200);
+    }
+
+
+    /**
+     * XML in test contains the same value as defined in: https://github.com/killbill/killbill/issues/1497, except that
+     * 'timeSinceEarliestUnpaidInvoiceEqualsOrExceeds' element is added
+     */
+    @Test(groups = "fast", description = "https://github.com/killbill/killbill/issues/1497")
+    public void testGetOverdueConfigJson_withTimeSinceEarliest() throws Exception {
```

Comment:
Naming convention: Same thing here. Typically we only do camel case and sometimes add `_` when name becomes too unreadable or when there is a need to add enum names with uppercase in the name (example [here](https://github.com/killbill/killbill/blob/master/beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestIntegrationDryRunInvoice.java#L726)